iceoryx_posh/source/popo/building_blocks/event_notifier.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include "iceoryx_posh/internal/popo/building_blocks/event_notifier.hpp"
+#include "iceoryx_posh/internal/log/posh_logging.hpp"
+
+namespace iox
+{
+namespace popo
+{
+EventNotifier::EventNotifier(EventVariableData& dataRef, const uint64_t index) noexcept
+    : m_pointerToEventVariableData(&dataRef)
+    , m_notificationIndex(index)
+{
+    if (index >= MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET)
+    {
+        LogWarn() << "The provided index " << index << " is too large. The index has to be in the range of [0, "
+                  << MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET << ").";
```

Comment:
Is this the right bracket? I'm not a mathematician but shouldn't this be `[0, MAX[`

Shouldn't this also be an error instead of a warning? It creates a unusable `EventNotifier`

Reply:
It is the right bracket, bracketman! But for the non mathematicians I will change it to ``[`` ;)